File: lib/core_services/comic/interfaces/comic_history.dart
import 'package:freezed_annotation/freezed_annotation.dart';

import '../main.dart';

part 'comic_history.freezed.dart';
part 'comic_history.g.dart';

@freezed
sealed class ComicHistory with _$ComicHistory {
  const factory ComicHistory({
    required String sourceId,
    required Comic item,
    required DateTime watchUpdatedAt,
    required ComicChapter lastChapter,
    required WatchPage watchPage,
  }) = _ComicHistory;

  factory ComicHistory.createFakeData() {
    return ComicHistory(
      sourceId: '',
      item: Comic.createFakeData(),
      watchUpdatedAt: DateTime.now(),
      lastChapter: ComicChapter.createFakeData(),
      watchPage: WatchPage(currentPage: 0, totalPage: 10),
    );
  }

  factory ComicHistory.fromJson(Map<String, dynamic> json) =>
      _$ComicHistoryFromJson(json);
}

File: lib/core_services/comic/interfaces/comic_history.ts
import type { Comic } from "./comic"
import type { ComicChapter } from "./comic_chapter"
import type { WatchPage } from "./watch_page"

export interface ComicHistory {
  sourceId: string
  item: Comic
  watchUpdatedAt: Date
  lastChapter: ComicChapter
  watchPage: WatchPage
}
